include/deepworks/optimizer.hpp

```diff
@@ -34,9 +35,35 @@ class SGDMomentum {
 private:
     float m_lr;
     float gamma;
+
     Parameters& m_params;
     std::vector<Tensor> velocities;
 };
 
+
+class Adam {
+public:
+    explicit Adam(Parameters& params, float lr, float beta_one = 0.9f,
+                  float beta_second = 0.999f, float epsilon = 0.001, size_t num_iterations = 0);
+
+    void step();
+
+    float get_lr() const;
+
+    void set_lr(float lr);
+
+private:
+    float m_lr;
+    float m_beta_one;
```

Comment:
https://pytorch.org/docs/stable/optim.html#torch.optim.Adam

According to pytorch it looks like a `std::array<float, 2>` as option. Honestly speaking I don't know, up to you